### DISCUSSION TITLE
how to query balance?

### DISCUSSION THREAD

**Discussion by ovicloud** [Chia-Network/chia-blockchain#5780] [Support]
how to query balance by address，for example：
 xch1sp9anlzjah8wf4nt4tn7278fh0uq6s8unvahfqdyagsrczekxwsstsjqtf

**Comment by mariano54** [comment]
get_coin_records_by_puzzle_hash, look at the WIKI for guide on RPC

**Reply by PrernaPadhee** [reply]
Hi, can you explain how to calculate the balance of the address by  using the coin records fetched by this RPC call ?

Consider the following response: 

The unspent coins for this puzzle hash is null and following are the spent coins. Upon checking the explorer the balance totals to 0 XCH. How does the coin amount total to 0 as per the following response ? Please elaborate.

$ curl --insecure --cert ~/.chia/mainnet/config/ssl/full_node/private_full_node.crt \--key ~/.chia/mainnet/config/ssl/full_node/private_full_node.key \-d '{"puzzle_hash":"b1736654875b1c49b4077b89580c4447f12f1e86fb85d488d7efddd5c6e06be0", "include_spent_coins": true}' \-H "Content-Type: application/json" -X POST [https://localhost:8555/get_coin_records_by_puzzle_hash](https://secure-web.cisco.com/18STtzn_aQf8fCkeGOoOSwZx6f22Vs8f6d1orbmbZAz1JUgPMbYWPFE7nTWYFacTMyO9k83qvaZGau8R-MCg4tGbHkfW4U6V1CJFjN85J77RHK8svzAht8xa5Vi4gpqY79Y2OTuVYQjygw9AEli0Cz7Upzuq0lnGRO6bSHJKauUCLpwXp3vkKdAqF1p_lFr9eo_ep_h3g21tBG0_bF1TST9WGdlAziCDe0-NhzT1AxryQW-XFjO0wq6Iw7aDZqaIYmaw_XN_UiTdSI4RvyH1Qw626fh2T2dVbnzbT3Gl6JKWYUY7HLCCkV-tMMkQQqMbSyTIbNuCe3S96Q-u4LAd_0zdrBLzHA94oG3dAF4mE_nTjQ3UQyXu8QYjBEYaKaUtUfUWHt7hVdbdgS8FNhrveYVg2PfAIN8xqt0Go6HwaYMHlCDuknqW0dJn9lHkBd_eTyEtpV1NDF4pD50L66qXnAOS0GqvDskHBNtjj2j4WK0bMJB_Yiu4JovjFODs3Thwy8teWjsu9Lx6z-QhC7al2eA/https%3A%2F%2Flocalhost%3A8555%2Fget_coin_records_by_puzzle_hash)  | python -m json.tool
{
    "coin_records": [
        {
            "coin": {
                "amount": 879220,
                "parent_coin_info": "0x2ecb52b31038404bc46d24b89233fa8bcb6a4761bf8c96c1868ee203300726bd",
                "puzzle_hash": "0xb1736654875b1c49b4077b89580c4447f12f1e86fb85d488d7efddd5c6e06be0"
            },
            "coinbase": false,
            "confirmed_block_index": 797728,
            "spent": true,
            "spent_block_index": 797778,
            "timestamp": 1630501702
        },
        {
            "coin": {
                "amount": 921800500,
                "parent_coin_info": "0x055c0286a0eb881f2e950ce4ee30870868d1a951a0e0eb4dc52de5472a9c8b91",
                "puzzle_hash": "0xb1736654875b1c49b4077b89580c4447f12f1e86fb85d488d7efddd5c6e06be0"
            },
            "coinbase": false,
            "confirmed_block_index": 922637,
            "spent": true,
            "spent_block_index": 922641,
            "timestamp": 1632832094
        }
    ],
    "success": true
}

**Reply by kimsk** [reply]
The balance is the total of unspent coins. Since all coins were spent, the total balance is zero.

**Reply by PrernaPadhee** [reply]
> The balance is the total of unspent coins. Since all coins were spent, the total balance is zero.

got it , thanks! And is there any way to fetch the coin records for the coins that this address would have received ?

there are inbound coins to the address -> https://www.chiaexplorer.com/blockchain/address/xch1k9ekv4y8tvwyndq80wy4srzyglcj785xlwzafzxhalwat3hqd0sqyggts4
how do i get this info via rpc calls ?

also, what does parent_coin_info indicate in a coin record ?

**Reply by kimsk** [reply]
You have to decode the address to `puzzle_hash` and use `get_coin_records_by_puzzle_hash` to look for the `coin records`. 

The `parent_coin_info` is the `coin_id` of the the coin that was spent and output the `CREATE_COIN` condition to create the new coin.

Feel free to ask more questions in https://developers.chia.net/.
